src/entity.ts

```diff
@@ -130,7 +150,48 @@ export namespace entity {
        * @name Int#value
        * @type {string}
        */
-      this.value = value.toString();
+      if (typeCastOptions) {
+        if (
+          typeCastOptions.integerTypeCastFunction &&
+          typeof typeCastOptions.integerTypeCastFunction !== 'function'
+        ) {
+          throw new Error(
+            `integerTypeCastFunction is not a function or was not provided.`
+          );
+        }
+        this.typeCastFunction = typeCastOptions.integerTypeCastFunction;
+        this.typeCastProperties = typeCastOptions.properties
```

Comment:
I think it's fair to just do `this.typeCastProperties = arrify(typeCastOptions.properties)`, even if it's empty 🤷‍♂